plugin/pkg/shoot/validator/admission.go

```diff
@@ -239,33 +240,55 @@ func (v *ValidateShoot) Admit(ctx context.Context, a admission.Attributes, _ adm
 		return apierrors.NewInternalError(fmt.Errorf("could not find referenced cloud profile: %+v", err.Error()))
 	}
 
-	var seed *core.Seed
+	coreCloudProfile := &core.CloudProfile{}
+	if err := gardencorev1beta1.Convert_v1beta1_CloudProfile_To_core_CloudProfile(cloudProfile, coreCloudProfile, nil); err != nil {
+		return apierrors.NewInternalError(fmt.Errorf("could not convert v1beta1 cloud profile: %+v", err.Error()))
+	}
+
+	var (
+		seed     *gardencorev1beta1.Seed
+		coreSeed *core.Seed
+	)
 	if shoot.Spec.SeedName != nil {
 		seed, err = v.seedLister.Get(*shoot.Spec.SeedName)
 		if err != nil {
 			return apierrors.NewInternalError(fmt.Errorf("could not find referenced seed %q: %+v", *shoot.Spec.SeedName, err.Error()))
 		}
+
+		coreSeed = &core.Seed{}
+		if err := gardencorev1beta1.Convert_v1beta1_Seed_To_core_Seed(seed, coreSeed, nil); err != nil {
+			return apierrors.NewInternalError(fmt.Errorf("could not convert v1beta1 seed: %+v", err.Error()))
+		}
 	}
 
-	project, err := admissionutils.ProjectForNamespaceFromInternalLister(v.projectLister, shoot.Namespace)
+	project, err := admissionutils.ProjectForNamespaceFromExternalLister(v.projectLister, shoot.Namespace)
 	if err != nil {
 		return apierrors.NewInternalError(fmt.Errorf("could not find referenced project: %+v", err.Error()))
 	}
 
-	var secretBinding *core.SecretBinding
+	var (
+		secretBinding     *gardencorev1beta1.SecretBinding
+		coreSecretBinding *core.SecretBinding
+	)
+
 	if a.GetOperation() == admission.Create && shoot.Spec.SecretBindingName != nil {
 		secretBinding, err = v.secretBindingLister.SecretBindings(shoot.Namespace).Get(*shoot.Spec.SecretBindingName)
 		if err != nil {
 			return apierrors.NewInternalError(fmt.Errorf("could not find referenced secret binding: %+v", err.Error()))
 		}
+
+		coreSecretBinding = &core.SecretBinding{}
+		if err := gardencorev1beta1.Convert_v1beta1_SecretBinding_To_core_SecretBinding(secretBinding, coreSecretBinding, nil); err != nil {
+			return apierrors.NewInternalError(fmt.Errorf("could not convert v1beta1 secret binding: %+v", err.Error()))
+		}
 	}
 
 	// begin of validation code
 	validationContext := &validationContext{
-		cloudProfile:  cloudProfile,
+		cloudProfile:  coreCloudProfile,
 		project:       project,
-		seed:          seed,
-		secretBinding: secretBinding,
+		seed:          coreSeed,
+		secretBinding: coreSecretBinding,
```

Comment:
Naming is bit inconsistent, Should we just remove the `core` prefix? project is also internal type but doesn't have the prefix.

Reply:
There is no conversion needed for project hence it does not have the prefix. For the other types conversion was needed so I named it with `core` prefix